katara/aci.py

```diff
@@ -117,10 +122,98 @@ def lookup_var(v: Expr) -> Expr:
         print("Actor is commutative")
 
 
+def check_i(filename: str, fnNameBase: str, loopsFile: str, cvcPath: str) -> None:
+    state_transition_analysis = analyze_new(
+        filename, fnNameBase + "_next_state", loopsFile
+    )
+
+    tracker = VariableTracker()
+
+    initial_state = tracker.variable(
+        "initial_state", state_transition_analysis.arguments[0].type
+    )
+
+    op_group = tracker.group("op")
+    op = [
+        op_group.variable(v.name(), v.type)
+        for v in state_transition_analysis.arguments[1:]
+    ]
+
+    afterState_op = tracker.variable(
+        "afterState_op", state_transition_analysis.arguments[0].type
+    )
+    afterState_op_op = tracker.variable(
+        "afterState_op_op", state_transition_analysis.arguments[0].type
+    )
+
+    vc = state_transition_analysis.call(initial_state, *op)(
+        tracker,
+        lambda obj0_after_op: Implies(
+            Eq(obj0_after_op, afterState_op),
+            state_transition_analysis.call(obj0_after_op, *op)(
+                tracker,
+                lambda obj0_after_op_op: Implies(
+                    Eq(obj0_after_op_op, afterState_op_op),
+                    Eq(obj0_after_op, obj0_after_op_op),
+                ),
+            ),
+        ),
+    )
+
+    toSMT(
+        [],
+        set(tracker.all()),
+        [],
+        [],
+        vc,
+        "./synthesisLogs/idempotence-test.smt",
+        [],
+        [],
+    )
+
+    procVerify = subprocess.run(
+        [
+            cvcPath,
+            "--lang=smt",
+            "--produce-models",
+            "--tlimit=100000",
+            "./synthesisLogs/idempotence-test.smt",
+        ],
+        stdout=subprocess.PIPE,
+    )
+
+    procOutput = procVerify.stdout
+    resultVerify = procOutput.decode("utf-8").split("\n")
+
+    def lookup_var(v: Expr) -> Expr:
```

Comment:
Can we extract this to a common function that takes both the variable and the `resultVerify`?

Reply:
Made the change!